test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1

```diff
@@ -122,5 +122,29 @@ namespace ModuleCmdlets
     $results[0] | Should Be $path
     $results[1] | Should BeExactly "BinaryModuleCmdlet1 exported by the ModuleCmdlets module."
     }
+
+    It "PS should try to load the assembly from assembly name if file path doesn't exist" {
+
+        if ($IsWindows)
+        {
+            New-ModuleManifest -Path $TESTDRIVE\test.psd1 -NestedModules \NOExistedPath\System.Management.Automation.dll
+        }
+        else
+        {
+            New-ModuleManifest -Path $TESTDRIVE/test.psd1 -NestedModules /NOExistedPath/System.Management.Automation.dll
+        }
```

Comment:
Can you use `Join-Path` to avoid the `if ($IsWindows) else` blocks? For example,
```
$psdFile = Join-Path $TESTDRIVE test.psd1
$nestedModule = Join-Path NOExistedPath System.Management.Automation.dll
New-ModuleManifest -Path $psdFile -NestedModules $nestedModule 
```

Reply:
resolved